### PR TITLE
fix add `on-contract-event`

### PR DESCRIPTION
**Pull request by juliangruber**
<del>Blocked by https://github.com/filecoin-station/on-contract-event/pull/3</del>

### CODE REVIEW

File: test/round-tracker.test.js [hunk 1]
```diff
@@ -221,8 +221,9 @@ describe('Round Tracker', () => {
   describe('startRoundTracker', () => {
     it('detects the current round', async function () {
       this.timeout(TIMEOUT_WHEN_QUERYING_CHAIN)
-      const { sparkRoundNumber, close } = await startRoundTracker(pgPool)
-      close()
+      const controller = new AbortController()
+      const { sparkRoundNumber } = await startRoundTracker(pgPool, controller.signal)
+      controller.signal.abort()
```
Comment (bajtos) [inline test/round-tracker.test.js:226]: This is not going to work. When `await startRoundTracker` throws an error, you will never reach the second line to abort the controller/signal.

I am proposing a slightly different implementation: https://github.com/filecoin-station/spark-api/pull/305

**Review by bajtos — CHANGES_REQUESTED** [review]

File: test/round-tracker.test.js (hunk 1 above)
Comment (juliangruber) [reply test/round-tracker.test.js:226]: Right! Will update once 305 lands 👍

File: lib/round-tracker.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import assert from 'node:assert'
 import * as Sentry from '@sentry/node'
-import { createMeridianContract } from './ie-contract.js'
+import { createMeridianContract, provider } from './ie-contract.js'
+import { onContractEvent } from 'on-contract-event'
+import { RPC_URL, GLIF_TOKEN } from '../spark-publish/ie-contract-config.js'
```
Comment (bajtos) [inline lib/round-tracker.js:5]: I'd like to keep only one place where we are importing from spark-publish.

Would you mind re-exporting these constants in `./ie-contract.js`?

```suggestion
import { RPC_URL, GLIF_TOKEN, createMeridianContract, provider } from './ie-contract.js'
import { onContractEvent } from 'on-contract-event'
```

**Review by bajtos — APPROVED** [review]
Feel free to change the imports in a follow-up pull request.

**Comment by bajtos** [comment]
We don't need this change anymore, see https://github.com/filecoin-station/spark-api/pull/306#issuecomment-2115449662

**Pull request closed without merging**